lib/sda/context.go

```diff
@@ -1,16 +1,17 @@
 package sda
 
-import (
-	"github.com/dedis/cothority/lib/network"
-)
+import "github.com/dedis/cothority/lib/network"
 
-// Context is the interface that is given to an Service
+// Context is the interface that is given to a Service
 type Context interface {
-	NewTreeNodeInstance(*Tree, *TreeNode) *TreeNodeInstance
+	NewTreeNodeInstance(*Tree, *TreeNode, string) *TreeNodeInstance
 	RegisterProtocolInstance(ProtocolInstance) error
 	SendRaw(*network.Entity, interface{}) error
+	CreateProtocolService(*Tree, string) (ProtocolInstance, error)
+	CreateProtocolSDA(*Tree, string) (ProtocolInstance, error)
```

Comment:
I saw those, but I thought the interface is the first thing one will look at. Anyways you are right, the (commented) implementations are just a few lines away...